godtools/App/Features/ToolTraining/ToolTrainingViewModel.swift

```diff
@@ -30,12 +31,12 @@ class ToolTrainingViewModel: NSObject, ToolTrainingViewModelType {
     let continueButtonTitle: ObservableValue<String> = ObservableValue(value: "")
     let numberOfTipPages: ObservableValue<Int> = ObservableValue(value: 0)
     
-    required init(flowDelegate: FlowDelegate, renderer: MobileContentRendererType, trainingTipId: String, tipModel: TipModelType, analytics: AnalyticsContainer, localizationServices: LocalizationServices, viewedTrainingTips: ViewedTrainingTipsService) {
+    required init(flowDelegate: FlowDelegate, pageRenderer: MobileContentPageRenderer, renderedPageContext: MobileContentRenderedPageContext, tipPages: [TipPage], trainingTipId: String, tipModel: Tip, analytics: AnalyticsContainer, localizationServices: LocalizationServices, viewedTrainingTips: ViewedTrainingTipsService) {
         
         self.flowDelegate = flowDelegate
-        self.renderer = renderer
-        self.resource = renderer.resource
-        self.language = renderer.language
+        self.renderedPageContext = renderedPageContext
+        self.tipPages = tipPages
```

Comment:
any reason you are passing in the `tipPages` array instead of just accessing it out of the `Tip` model which you already have?

Reply:
Hey @frett good catch, I missed that one.